### PR TITLE
Views

### PR DESCRIPTION
**Pull request by LuisMenjivar**

### CODE REVIEW

**Comment by eliotsykes** [comment]
Try to use unique names for branches that are descriptive. Same goes for Pull Request titles. PR6 + PR7  had the same branch names and titles that didn't describe the work done accurately which has made reviewing them confusing.

File: spec/features/bookmarks_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require "rails_helper"
+feature "boomarks" do 
+  before(:each) do 
+    @user = create(:user)
+    @topic = create(:topic, user: @user)
+    @bookmark = create(:bookmark, topic: @topic)
+    manual_sign_in(@user)
+  end
+  scenario "user can see boomarks index" do
```
Comment (eliotsykes) [inline spec/features/bookmarks_spec.rb:9]: typo boo**k**marks

File: spec/features/bookmarks_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+feature "boomarks" do 
+  before(:each) do 
+    @user = create(:user)
+    @topic = create(:topic, user: @user)
+    @bookmark = create(:bookmark, topic: @topic)
+    manual_sign_in(@user)
+  end
+  scenario "user can see boomarks index" do 
+    visit topic_path(@topic)
+    expect(page).to have_css("h4", text: "#{@topic.bookmarks.first.url}")
```
Comment (eliotsykes) [inline spec/features/bookmarks_spec.rb:11]: String interpolation needed here in the `text` option?

File: spec/features/bookmarks_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    fill_in "bookmark_url", with: "www.google.com"
+    click_button("Create Bookmark")
+    visit topic_path(@topic)
+    expect(page).to have_css("h4", text: "www.google.com")
+  end
+
+  scenario "user can view a boomark" do 
+    visit topic_path(@topic)
+    click_link("Show")
+    expect(page).to have_css("h4", text: "#{@topic.bookmarks.first.url}")
```
Comment (eliotsykes) [inline spec/features/bookmarks_spec.rb:25]: Check string interpolation is needed. Try without.

File: spec/features/topics_spec.rb [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
     manual_sign_in(@user)
- end
+  end
 
   scenario "User can browse through an index of his topics" do 
-    topic = create(:topic, user_id: nil)
+    topic = create(:topic, user: nil)
     visit topics_path
     expect(page).to have_css("h1", text: "Topics") 
     expect(page).not_to have_css("h4", text: "#{topic.title}")
```
Comment (eliotsykes) [inline spec/features/topics_spec.rb:13]: Unnecessary string interpolation on lines 13, 16, 24.

File: spec/features/topics_spec.rb [hunk 5]
```diff
@@ -28,9 +28,29 @@
...
+    expect(page).to have_text "New topic successfully created!"
     expect(page).to have_css("h4", text: "animals")
   end
+
+  scenario "user can edit a topic" do 
+    topic = create(:topic, title: "animals", user: @user)
+    visit topics_path
+    click_link("animals")
+    click_link("Edit Topic")
+    fill_in "topic_title", with: "wild animals"
```
Comment (eliotsykes) [inline spec/features/topics_spec.rb:40]: Add expectation to test that the topic_title text field contains the old value "animals" before changing it to "wild animals".

File: app/views/topics/show.html.erb [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
+  </div>
+  <div class="col-md-3">
+    <h1><%= link_to "New Bookmark", new_topic_bookmark_path(@topic, @post), class: "btn btn-block btn-lg btn-primary" %></h1>
+  </div>
+</div>
 <div class="row">
   <div class="col-md-8">
-    <% @topic_bookmarks.each do |topic_bookmark| %>
+    <% @bookmarks.each do |topic_bookmark| %>
       <% @url = "#{topic_bookmark.url}" %>
```
Comment (eliotsykes) [inline app/views/topics/show.html.erb:18]: Is `@url` needed?

**Comment by eliotsykes** [comment]
Luis, it'd be good to address the comments I've left on the pull request before adding any more features to the app.

File: app/views/topics/show.html.erb (hunk 6 above)
Comment (LuisMenjivar) [inline app/views/topics/show.html.erb:18]: I am iterating over all the bookmark urls for that given topic. Also, I removed the string interpolation, I changed it from @url = "#{topic_bookmark.url}" to: url = topic_bookmark.url  . I needed the url for the href  and for the text shown with in the link < a href= <%= url %> >  <%= url %>  </a>

**Pull request merged**